packages/core/src/events/input/promptTemplate/base.ts

```diff
@@ -0,0 +1,123 @@
+import { Callable, CallableConfig } from "../../../record/callable";
+import { variableValidator } from "../../../utils/promptTemplateValidator/variableValidator";
+import { BaseEvent, BaseEventParams } from "../../base";
+export interface PromptTemplateParams extends BaseEventParams {
+    /**
+     * Clearly declare the input variables inside the template
+     */
+    inputVariables?: string[] | undefined;
+
+    /**
+     * the template string
+     */
+    template?: string | "";
+
+    validator?: variableValidator | undefined;
+
+    partialVariables?: string[] | undefined;
+}
+
+export class basePromptTemplate <
```

Comment:
For the class naming convention in TypeScript, you need to capitalize the first character of the class name. So the name should be `BasePromptTemplate`.